tgui/packages/tgui/interfaces/Autolathe.js

```diff
@@ -0,0 +1,272 @@
+        {Object.keys(materials).map((category) => (
+          <Flex.Item key={category} grow>
+            <ProgressBar
+              width="99%"
+              value={materials[category] / capacity[category]}>
+              <Box textAlign="center">
+                {capitalize(category)}: {materials[category]}/
+                {capacity[category]}
+              </Box>
+            </ProgressBar>
+          </Flex.Item>
+        ))}
+      </Flex>
+      <CurrentlyMaking />
+    </Section>
+  );
+};
+
+const CurrentlyMaking = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { currently_making } = data;
+
+  return currently_making ? (
+    <Fragment>
+      <Box height="5px" />
+      <Button
+        fluid
+        textAlign="center"
+        content={
+          'Currently making: ' +
+          capitalize(currently_making.name) +
+          (currently_making.multiplier > 1
+            ? ' (x' + currently_making.multiplier + ')'
+            : '')
+        }
+      />
+    </Fragment>
+  ) : null;
+};
+
+const QueueList = (props, context) => {
+  const { act, data } = useBackend(context);
+  const {
+    materials,
+    capacity,
+    queued,
+    printables,
+    selectable_categories,
+    queuemax,
+  } = data;
+
+  return (
+    <Section title="Queue">
+      <Flex direction="column">
+        {queued.map((item) => {
+          return (
+            <Flex.Item key={item}>
+              <Flex direction="row">
+                <Flex.Item>
+                  <Button
+                    content={
+                      item.index +
+                      ': ' +
+                      capitalize(item.name) +
+                      (item.multiplier > 1 ? ' (x' + item.multiplier + ')' : '')
+                    }
+                  />
+                  <Box width="5px" />
+                </Flex.Item>
+                <Flex.Item>
+                  <Button
+                    icon="xmark"
+                    tooltip="Remove"
+                    onClick={() =>
+                      act('cancel', {
+                        index: item.index,
+                        name: item.name,
+                        multiplier: item.multiplier,
+                      })
+                    }
+                  />
+                </Flex.Item>
+              </Flex>
+            </Flex.Item>
+          );
+        })}
+      </Flex>
+    </Section>
+  );
+};
+
+// the below all has to be in one section due to the categories and search params
+const PrintablesSection = (props, context) => {
+  const { act, data } = useBackend(context);
+
+  const logger = createLogger('autolathe');
+
+  const {
+    materials,
+    capacity,
+    queued,
+    printables,
+    selectable_categories,
+    queuemax,
+  } = data;
+
+  const [currentSearch, setSearch] = useLocalState(
+    context,
+    'current_search',
+    ''
+  );
+
+  const categories = [];
+  for (let i = 0; i < printables.length; i++) {
```

Comment:
You don't really want to use for loops unless absolutely necessary, this should be done using a filter and map

`printables.filter(x => categories.includes(x.recipe_category)).map(x => x.recipe_category)`